From 9a19ec0fe62ac7ea641cc8e215ef1e1faaf508b5 Mon Sep 17 00:00:00 2001
From: Aelin <aelin@postmarketos.org>
Date: Sat, 12 Sep 2026 15:09:58 +0200
Subject: [PATCH 09/10] systemd: allow reading nsresourced socket xattrs

systemd's userdb client reads user.userdb.* extended attributes from
the nsresourced socket before connecting to it, to determine whether
the backend can satisfy a lookup.

Allow the existing nsresourced stream-connect interface to read the
socket inode.

This was introduced in systemd commit 00a7b1d2d8d5de7548da69123832b913467235c0
---
 policy/modules/system/systemd.if | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/policy/modules/system/systemd.if b/policy/modules/system/systemd.if
index 067cae1cc..8e81689bc 100644
--- a/policy/modules/system/systemd.if
+++ b/policy/modules/system/systemd.if
@@ -3338,7 +3338,7 @@ interface(`systemd_nsresourced_stream_connect', `
 	')
 
 	allow $1 systemd_nsresourced_t:unix_stream_socket connectto;
-	allow $1 systemd_nsresourced_runtime_t:sock_file write_sock_file_perms;
+	allow $1 systemd_nsresourced_runtime_t:sock_file rw_sock_file_perms;
 ')
 
 #######################################
-- 
2.55.0

